src/ecs/resources/input.rs

```diff
@@ -335,30 +248,66 @@ impl InputHandler {
     /// this method can return None, either if no mouse is connected, or if no mouse events have
     /// been recorded
     pub fn mouse_position(&self) -> Option<(i32, i32)> {
-        self.current_frame.mouse_position
+        self.mouse_position
     }
 
     /// Gets the change in position since the last frame.
     pub fn mouse_position_change(&self) -> (i32, i32) {
-        match (self.current_frame.mouse_position, self.previous_frame.mouse_position) {
+        match (self.mouse_position, self.previous_mouse_position) {
             (Some(current), Some(previous)) => (current.0 - previous.0, current.1 - previous.1),
             _ => (0, 0),
         }
     }
 
-    /// Returns an iterator over the buttons that are currently pressed
-    pub fn pressed_buttons(&self) -> PressedButtons {
-        self.current_frame.pressed_buttons()
+    /// Checks if all the given mouse buttons are being pressed.
+    pub fn mouse_buttons_are_pressed(&self, buttons: &[MouseButton]) -> bool {
+        buttons.iter().all(|btn| self.mouse_button_is_pressed(*btn))
+    }
+
+    /// Returns a vector containing the buttons that are currently pressed
+    pub fn pressed_buttons(&self) -> ButtonIterator {
+        let mouse_buttons = self.pressed_mouse_buttons
+            .iter()
+            .map((|&mb| Button::Mouse(mb)) as fn(&MouseButton) -> Button);
+        let keys = self.pressed_keys.iter().map((|&k| Button::Key(k)) as fn(&VirtualKeyCode) -> Button);
+        ButtonIterator {
+            iterator: mouse_buttons.chain(keys)
+        }
+    }
+
+    /// Returns a vector containing the buttons that were pressed this frame
+    pub fn down_buttons(&self) -> ButtonIterator {
```

Comment:
Could `ButtonIterator` be renamed to `Buttons` to remain consistent with other iterator names?